File: src/dataset/go/sublist_Exercism_go_rootulp/openAI/ChatGPT4_sublist_v1.go
package sublist

type Relation string

func Sublist(a []int, b []int) Relation {
	switch {
	case len(a) == len(b) && equal(a, b):
		return "equal"
	case len(a) < len(b) && isSublist(a, b):
		return "sublist"
	case len(a) > len(b) && isSublist(b, a):
		return "superlist"
	default:
		return "unequal"
	}
}

// isSublist returns true if a is a sublist of b
func isSublist(a []int, b []int) bool {
	if len(a) == 0 {
		return true
	}
	for i := 0; i <= len(b)-len(a); i++ {
		if equal(a, b[i:i+len(a)]) {
			return true
		}
	}
	return false
}

// equal checks if two slices are equal
func equal(a, b []int) bool {
	if len(a) != len(b) {
		return false
	}
	for i := range a {
		if a[i] != b[i] {
			return false
		}
	}
	return true
}

File: src/dataset/typescript/sublist_Exercism-typescript-FilipeCerejo/openAI/ChatGPT4_sublist_v1.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.List = void 0;
class List {
    constructor(...list) {
        this.list = list;
    }
    compare(otherList) {
        const len1 = this.list.length;
        const len2 = otherList.list.length;
        if (len1 === len2) {
            return this.isEqual(this.list, otherList.list) ? 'equal' : 'unequal';
        }
        if (len1 > len2) {
            return this.isSublist(otherList.list, this.list) ? 'superlist' : 'unequal';
        }
        return this.isSublist(this.list, otherList.list) ? 'sublist' : 'unequal';
    }
    isSublist(smaller, bigger) {
        const smallerLen = smaller.length;
        const biggerLen = bigger.length;
        if (smallerLen === 0)
            return true;
        for (let i = 0; i <= biggerLen - smallerLen; i++) {
            if (this.isEqual(bigger.slice(i, i + smallerLen), smaller)) {
                return true;
            }
        }
        return false;
    }
    isEqual(arr1, arr2) {
        if (arr1.length !== arr2.length)
            return false;
        for (let i = 0; i < arr1.length; i++) {
            if (arr1[i] !== arr2[i])
                return false;
        }
        return true;
    }
}
exports.List = List;
